hw06-atm-emulator/src/main/java/ru/otus/Main.java

```diff
@@ -0,0 +1,44 @@
+package ru.otus;
+
+/*
+    java -jar atm-emulator.jar
+*/
+
+import ru.otus.Exceptions.CashBoxException;
+import ru.otus.Exceptions.DispenserException;
+import ru.otus.Builders.AtmBuilder;
+import ru.otus.Builders.DispenserBuilder;
+import ru.otus.Builders.PrinterBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Main {
+    private static final List<BanknoteBatch> banknoteBatches = new ArrayList<BanknoteBatch>() {{
+        add(new BanknoteBatch(100, 100));
+        add(new BanknoteBatch(200, 50));
+        add(new BanknoteBatch(500, 1));
+        add(new BanknoteBatch(1000, 5));
+        add(new BanknoteBatch(5000, 2));
+    }};
+
+    public static void main(String[] args) throws DispenserException, CashBoxException {
+        var configuration = new Configuration(Arrays.asList(100, 200, 500, 1000, 5000));
+        var dispenserFactory = new DispenserBuilder(configuration);
+        var printerFactory = new PrinterBuilder(configuration);
+
+        var atm = new AtmBuilder(configuration,
+                                    dispenserFactory,
+                                    printerFactory)
+                        .WithDispenser()
+                        .WithPrinter()
+                        .Build();
```

Comment:
Здесь логичнее выглядело бы так:
```suggestion
        var atm = new AtmBuilder(configuration)
                        .WithDispenser(dispenserFactory)
                        .WithPrinter(printerFactory)
                        .Build();
```

А точнее даже так:
```suggestion
        var atm = new AtmBuilder(configuration)
                        .WithDispenserFactory(dispenserFactory)
                        .WithPrinterFactory(printerFactory)
                        .Build();
```

Или передавать не фабрики, а конкретные объекты.